### PR TITLE
Extract package pandoc-core

### PR DESCRIPTION
**Pull request by tarleb**
Closes: #6215

### CODE REVIEW

**Comment by tarleb** [comment]
I'm not sure about this. Having this would allow to write readers and writers without having to depend on the full pandoc lib. It could be helpful for some projects like pandoc-crossref, but I'm worried about having to maintain too many packages.

File: pandoc-core/pandoc-core.cabal [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+                     , Text.Pandoc.Parsing.Capabilities
+                     , Text.Pandoc.Parsing.Citations
+                     , Text.Pandoc.Parsing.General
+                     , Text.Pandoc.Parsing.GridTable
+                     , Text.Pandoc.Parsing.Lists
+                     , Text.Pandoc.Parsing.Math
+                     , Text.Pandoc.Parsing.Smart
+                     , Text.Pandoc.Parsing.State
+                     , Text.Pandoc.Parsing.Types
+                     , Text.Pandoc.Readers.LaTeX.Types
```
Comment (jgm) [inline pandoc-core/pandoc-core.cabal:119]: Why this?  It seems LaTeX-specific and thus not a candidate for core, but I assume something else here depends on it?

**Comment by jgm** [comment]
I'm leaning to the view that we should not do this; I don't see clear advantages to doing it that would outweigh the disadvantages.  The principle of the split -- what goes in core and what doesn't -- isn't conceptually clean.  E.g., XML goes one way, emojis go the other; citeproc goes one way, highlighting another; MIME goes one way, image size another -- and it's not clear why one of these is more "core" than another.  (I don't mean to say this selection is arbitrary -- presumably it's a kind of minimum core that can be cleanly extracted -- but that the principle of selection won't be obvious.

I'm open to persuasion though.

File: pandoc-core/pandoc-core.cabal (hunk 1 above)
Comment (tarleb) [reply pandoc-core/pandoc-core.cabal:119]: `Macro` from that module is uesd by Text.Pandoc.Parsing.{State,Capabilities}. There's probably a better solution, this is all very experimental.

**Comment by tarleb** [comment]
I had this code stashed away in my local repo for some time. My main objective back then was to have `Text.Pandoc.Parsing` in a separate package, the selection of modules was driven by that. I think I explored this while writing the jira package, but not sure.

I agree that the downsides out-weight the benefits. Closing...

**Pull request closed without merging**